src/filetree.rs

```diff
@@ -280,12 +280,63 @@ pub(crate) fn syncfs(d: &openat::Dir) -> Result<()> {
     rustix::fs::syncfs(d).map_err(Into::into)
 }
 
+fn copy_dir(src: &Path, dst: &Path) -> Result<()> {
+    let r = std::process::Command::new("cp")
+        .args(["-a"])
+        .arg(src)
+        .arg(dst)
+        .status()?;
+    if !r.success() {
+        anyhow::bail!("Failed to copy");
+    }
+    Ok(())
+}
+
+/// Convert path parent to temp, `fedora/foo` -> `fedora.tmp/foo`
 #[cfg(any(target_arch = "x86_64", target_arch = "aarch64"))]
-fn tmpname_for_path<P: AsRef<Path>>(path: P) -> std::path::PathBuf {
+fn tmp_for_path<P: AsRef<Path>>(path: P) -> std::path::PathBuf {
     let path = path.as_ref();
-    let mut buf = path.file_name().expect("filename").to_os_string();
-    buf.push(TMP_PREFIX);
-    path.with_file_name(buf)
+    let mut parent = String::from("");
+    if let Some(p) = path.parent() {
+        parent = format!("{}.tmp", p.display());
+    }
+    let buf = path.file_name().expect("filename").to_os_string();
+    Path::new(parent.as_str()).join(buf)
+}
+
+/// Get diff based on sub dir
+#[cfg(any(target_arch = "x86_64", target_arch = "aarch64"))]
```

Comment:
Also, while this is not at all directly related to your change, I found all the copy-paste of this conditional (there's a ton! you're only adding a bit more) in this file to be pretty ugly. I did https://github.com/coreos/bootupd/pull/673 in prep for reworking things here.

Probably the simplest thing to do is just to build this on all architectures and slap an `#[allow(dead_code)]` on the whole thing. But that's a distinct change.